### PR TITLE
Use Jena datatype equality

### PR DESCRIPTION
**Pull request by langsamu**

### CODE REVIEW

File: src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java [hunk 1]
```diff
@@ -308,7 +303,7 @@ private void closeAndThrow(final Model model, final String message) throws Kettl
     }
 
     private Object convertSqlValueToRdf(final Object sqlValue, @Nullable final RDFDatatype rdfDatatype) {
-        if (rdfDatatype == null || rdfDatatype.getURI().equals(XSD_NAMESPACE_IRI + "#string")) {
+        if (rdfDatatype == null || rdfDatatype.equals(XSDDatatype.XSDstring)) {
```
Comment (adamretter) [inline src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java:306]: @langsamu It's a nice idea. But... have you tested this?

I am concerned that `RDFDatatype` does not override the object equality method `equals` so you end up with the default from `Object#equals(Object)` which is instead instance equality i.e. `a == b`.

As long as there is only a single instance of each RDFDatatype in the JVM then that will be fine... but I am not certain that is the case.

Comment (langsamu) [reply src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java:306]: That `rdfDatatype` comes from `addResourceProperties` via [`TypeMapper`](https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/datatypes/TypeMapper.html), "a global registry of known datatypes" based on a (URI) string hash:
https://github.com/adamretter/kettle-jena-plugins/blob/fee4214032c2d1f1b8b5b052f002e08a312e6685/src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java#L286

TypeMapper loads XSD simple types
https://github.com/apache/jena/blob/9fb9f050090c8fabbcca9552ab98a517aae78449/jena-core/src/main/java/org/apache/jena/datatypes/TypeMapper.java#L72

using the same `XSDDatatype` we're comparing to
https://github.com/apache/jena/blob/9fb9f050090c8fabbcca9552ab98a517aae78449/jena-core/src/main/java/org/apache/jena/datatypes/xsd/XSDDatatype.java#L589

so yes, they're the same.

File: src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java [hunk 2]
```diff
@@ -303,7 +303,8 @@ private void closeAndThrow(final Model model, final String message) throws Kettl
     }
 
     private Object convertSqlValueToRdf(final Object sqlValue, @Nullable final RDFDatatype rdfDatatype) {
-        if (rdfDatatype == null || rdfDatatype.equals(XSDDatatype.XSDstring)) {
+        // Using reference equality because TypeMapper guarantees uniqueness in caller
+        if (rdfDatatype == null || rdfDatatype == XSDDatatype.XSDstring) {
```
Comment (adamretter) [inline src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java]: I actually think directly using reference equality is a bad idea in-case the API changes. At the moment calling .equals will use their API which ultimately does reference equality.

File: src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java (hunk 1 above)
Comment (adamretter) [reply src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java:306]: Okay sounds good.

File: src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/EndToEndIT.java [hunk 3]
```diff
@@ -79,8 +77,8 @@ public void createXMLLiteral(@TempDir final Path tempDir) throws KettleException
         final Model expected = ModelFactory.createDefaultModel();
         expected.add(
                 expected.createResource("http://example.com/s"),
-                expected.createProperty(DCT_NAMESPACE_IRI + "description"),
-                expected.createTypedLiteral("<greeting>Hello <b>World</b>!</greeting>", XMLLiteralType.theXMLLiteralType));
+                DCTerms.description,
```
Comment (adamretter) [inline src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/EndToEndIT.java:80]: Nice!

**Review by adamretter — APPROVED** [review]
LGTM

**Pull request merged**